docs/sources/developers/kinds/maturity.md

```diff
@@ -0,0 +1,403 @@
+| Composable Kind<br/>Guide &#124; [Spec](https://github.com/grafana/grafana/blob/fb42532a6c49408a5da940586d4b053a5e8e752b/pkg/framework/kind/type.cue#L126-L140)                                                                                                    | Schemas defined in plugins that adhere to a meta-schema specified by a [Grafana slot](https://docs.google.com/document/d/13Rv395_T8WTLBgdL-2rbXKu0fx_TW-Q9yz9x6oBjm6g/edit#heading=h.f3onnwdi0a4d), and are intended for composition into a core or custom kind.These lack any functional maturity process (for now). | Merged<br/>Experimental<br/>Stable<br/>Mature | [barchart](https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/barchart/models.cue)                                                                                                                                                                            |
+
+Consistent with the common industry pattern of wrapping a DSL around schemas, each of these categories is defined in CUE
+by the Kind framework. Each definition specifies the way a Kind is declared in a .cue file on disk, and is linked as 
+“Spec” in the above table.
+
+The properties govern certain code generation and runtime behavior for the Kind, and controls how all of these behaviors
+relate to the path to maturity.
+
+## Maturity process guides
+
+Grafana’s schematized Kinds are all expressed using Thema. Consequently, all the schematization cases have a common 
+first step: writing schemas with Thema. It is recommended to complete 
+[Thema’s quickstart](https://github.com/grafana/thema/blob/main/docs/go-quickstart.md), which steps through schema 
+authoring basics.
+
+All have in common that there is one .cue input file, and one command to run that updates all the targets. Our code 
+generators are idempotent - they 
+[*never* produce stubs](https://github.com/grafana/grafana/pull/54816#issuecomment-1256341790). Consequently, it is a 
+system you can learn a lot about through trial and error: try something, run the generator, then see the effects by 
+running `git diff`.
+
+When writing Thema schemas within Grafana, there are a few key differences vs. the approach in the Thema tutorial:
+- All schema-based codegen in grafana/grafana is performed by make `gen-cue`, rather than by invoking the thema CLI.
+- Grafana code generators support certain attributes the Thema CLI is not natively aware of.
+- Thema lineages are expected to be declared at specific subpaths within .cue files, varying by category.
+
+## Schema-writing guidelines
```

Comment:
Let's separate these out into their own doc in a follow-up PR

Reply:
Agreed, in the follow-up PRs. For the moment I pushed it below the Milestones section. Hope it's fine